sympy/liealgebras/type_e.py

```diff
@@ -47,34 +47,43 @@ def simple_root(self, i):
         element of Q can be written as a linear combination
         of elements of D with all coefficients non-negative.
 
-        This method returns the ith simple root for E_n.
+        This method returns the ith simple root for E_n. Simple
+        roots are chosen in the basis that allows for easy subgrouping
+        of E6 and E7 from E8 simple roots.
+
+        Sources
+        =======
+        - https://en.wikipedia.org/wiki/E8_(mathematics)#Simple_roots
+        - https://en.wikipedia.org/wiki/E7_(mathematics)#Root_system
+        - https://en.wikipedia.org/wiki/E6_(mathematics)#Roots_of_E6
 
         Examples
         ========
 
         >>> from sympy.liealgebras.cartan_type import CartanType
         >>> c = CartanType("E6")
         >>> c.simple_root(2)
-        [1, 1, 0, 0, 0, 0, 0, 0]
+        [0, 1, -1, 0, 0, 0]
+
         """
         n = self.n
-        if i == 1:
-            root = [-0.5]*8
-            root[0] = 0.5
-            root[7] = 0.5
-            return root
-        elif i == 2:
-            root = [0]*8
-            root[1] = 1
-            root[0] = 1
-            return root
+        e = eye(8)
+
+        if i == n:
+            _root =  e.row(n-3) - e.row(n-2)
```

Comment:
There is no need for a leading underscore. Local variables are automatically private.